dagger/pipeline/tasks/spark_task.py

```diff
@@ -16,18 +14,25 @@ def init_attributes(cls, orig_cls):
                 Attribute(
                     attribute_name="spark_engine",
                     parent_fields=["task_parameters"],
-                    required=False,
-                    comment="Where to run spark job. Accepted values: emr/batch",
+                    required=True,
+                    comment="Where to run spark job. Accepted values: emr, batch, glue",
                 ),
-                Attribute(attribute_name="job_file", parent_fields=["task_parameters"]),
+                Attribute(attribute_name="job_file", parent_fields=["task_parameters"],  required=True),
+                Attribute(attribute_name="cluster_name", parent_fields=["task_parameters"],  required=True),
```

Comment:
Are you sure this should be required? In glue we are not using it. Maybe you could check in the __init__() that in case of engine is emr this shouldn't be null. And if it is throw the dagger missing parameter exception.